hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2792,7 +2792,32 @@ public OmBucketInfo getBucketInfo(String volume, String bucket)
             bucket, null);
       }
       metrics.incNumBucketInfos();
-      return bucketManager.getBucketInfo(volume, bucket);
+
+      ResolvedBucket resolvedBucket =
+          resolveBucketLink(Pair.of(volume, bucket), true);
```

Comment:
Even bucket is not `linkBucket` still we are resolving bucket, that looks like is not necessary. I believe all these operations which is currently been done inside `resolvedBucketLink` is required only for link buckets. In the old code it just skips doing for link buckets.